documentation/src/main/asciidoc/user_guide/clustered_lock.adoc

```diff
@@ -0,0 +1,200 @@
+node can release the lock at a given time.
+
+* `INSTANCE` - not yet supported
+
+When a `ClusteredLock` is defined, this lock can be used from all the nodes in the infinispan cluster.
+When the ownership is INSTANCE type, this means that the owner of the lock is the actual instance we acquired when
+`ClusteredLockManager.get("lockName")` is called.
+
+This means that each time we get a `ClusteredLock` instance with the `ClusteredCacheManager`, this instance will be
+a new instance.
+This lock can be used to synchronize code between infinispan nodes and inside each infinispan node.
+The advantage of this lock is that only the instance that called 'lock' can release the lock.
+
+
+==== Reentrancy
+
+When a `ClusteredLock` is configured reentrant, this means the owner of the lock can reacquire the lock as many consecutive
+times as it wants while holding the lock.
+
+Currently, only non reentrant locks are supported. This means that when two consecutive `lock` calls are sent for the same
+owner, the first call will acquire the lock if it's available, and the second call will block.
+
+
+=== The ClusteredLockManager interface.
+
+The `ClusteredLockManager` interface, *marked as experimental*, is the entry point to define, retrieve and remove a lock.
+It automatically listen to the creation of `EmbeddedCacheManager` and proceeds with the registration  of an
+instance of it per `EmbeddedCacheManager`.
+It starts the caches needed to store the lock state.
+
+Retrieving the `ClusteredLockManager` is as simple as invoking the `EmbeddedClusteredLockManagerFactory.from(EmbeddedCacheManager)`
+as shown in the example below:
+
+[source,java]
+----
+// create or obtain your EmbeddedCacheManager
+EmbeddedCacheManager manager = ...;
+
+// retrieve the ClusteredLockManager
+ClusteredLockManager clusteredLockManager = ClusteredLockManagerFactory.from(manager);
+----
+
+
+[source,java]
+----
+@Experimental
+public interface ClusteredLockManager {
+
+   boolean defineLock(String name);
+
+   boolean defineLock(String name, ClusteredLockConfiguration configuration);
+
+   ClusteredLock get(String name);
+
+   ClusteredLockConfiguration getConfiguration(String name);
+
+   boolean isDefined(String name);
+
+   CompletableFuture<Boolean> remove(String name);
+
+   CompletableFuture<Boolean> forceRelease(String name);
+}
+----
+
+* `defineLock` :
+Defines a lock with the specific name and the default `ClusteredLockConfiguration`. It does not overwrite
+existing configurations.
+
+* `defineLock(String name, ClusteredLockConfiguration configuration)` :
+Defines a lock with the specific name and `ClusteredLockConfiguration`. It does not overwrite existing
+configurations.
+
+* `ClusteredLock get(String name)` :
+Get’s a `ClusteredLock` by it’s name. A call of `defineLock` must be done at least once in the cluster.
+See ownership level section to understand the implications of `get` method call.
```

Comment:
Provide an anchor to the section you mention

Reply:
ok